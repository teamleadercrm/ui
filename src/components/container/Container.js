import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from '../box';
import cx from 'classnames';
import theme from './theme.css';

/** @type {React.ComponentType<any>} */
class Container extends PureComponent {
  render() {
    const { children, className, fixed, ...others } = this.props;

    const classNames = cx(
      theme['container'],
      {
        [theme['is-fixed']]: fixed,
      },
      className,
    );

    return (
      <Box {...others} boxSizing="content-box" className={classNames}>
        {children}
      </Box>
    );
  }
}

Container.propTypes = {
  children: PropTypes.node,
  fixed: PropTypes.bool,
};

export default Container;
