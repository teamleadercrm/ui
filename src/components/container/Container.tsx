import React, { ReactNode } from 'react';
import Box from '../box';
import cx from 'classnames';
import theme from './theme.css';
import { BoxProps } from '../box/Box';

interface Props extends Omit<BoxProps, 'ref'> {
  children?: ReactNode;
  className?: string;
  fixed?: boolean;
}

const Container = ({ children, className, fixed, ...others }: Props) => {
  const classNames = cx(
    theme['container'],
    {
      [theme['is-fixed']]: fixed,
    },
    className,
  );

  return (
    <Box {...others} boxSizing="content-box" className={classNames}>
      {children}
    </Box>
  );
};

export default Container;
