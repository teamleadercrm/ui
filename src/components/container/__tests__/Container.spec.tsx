import React from 'react';
import { render } from '@testing-library/react';
import Container from '../Container';

describe('Component - Container', () => {
  it('renders', async () => {
    const { asFragment } = render(<Container>Foobar</Container>);
    expect(asFragment()).toMatchSnapshot();
  });
});
