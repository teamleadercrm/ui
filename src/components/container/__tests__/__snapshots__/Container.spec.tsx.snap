// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Container renders 1`] = `
<DocumentFragment>
  <div
    class="box container"
    data-teamleader-ui="container"
    style="box-sizing: content-box;"
  >
    Foobar
  </div>
</DocumentFragment>
`;
